### ISSUE TITLE
Contrib Group Application:

### ISSUE THREAD

**Issue by skynetindia** [backdrop-ops/contrib#794]
Hello Backdrop Team,
 
Greetings from Skynet Technologies USA LLC.
 
We developed our All in One Accessibility module on Backdrop and want to publish it with the Backdrop modules list repo.
 
The All in One Accessibility is an AI-based accessibility tool, it improves website WCAG 2.2 ADA and other Global compliance standards and helps the Blind, Motor Impaired, Visually Impaired, Color Blind, Dyslexia, Cognitive and learning, Seizure & Epileptic, ADHD etc., user access website / Digital content. It supports over 140 languages and has Addons for Document / PDF Compliance remediation, Language translation as per accessibility user needs, and other Customization.

Please find the required details for joining the Contrib group.

**Please indicate how you intend to help the Backdrop community by joining this group**
* Option 1: I would like to contribute a project
* Option 2: I would like to maintain a project, but have nothing to contribute at this time
* Option 3: I would like to update documentation and/or triage issue queues

Option 1

## Based on your selection above, please provide the following information:

**(option 1) The name of your module, theme, or layout**
All in One Accessibility

## (option 1) Please note these 3 requirements for new contrib projects:

- [x] Include a README.md file containing license and maintainer information.
      You can use this example: https://github.com/skynettechnologies/backdrop_allinoneaccessibility/blob/main/README.md
- [x] Include a LICENSE.txt file.
      You can use this example: https://github.com/skynettechnologies/backdrop_allinoneaccessibility/blob/main/LICENSE.txt.
- [ ] If porting a Drupal 7 project, Maintain the Git history from Drupal.

**(option 1 -- optional) Post a link here to an issue in the drupal.org queue notifying the Drupal 7 maintainers that you are working on a Backdrop port of their project**

**Post a link to your new Backdrop project under your own GitHub account (option 1)**

https://github.com/skynettechnologies/backdrop_allinoneaccessibility

**Comment by stpaultim** [comment]
@skynetindia  Sorry for the lack of response. I'll see if we can't get someone to follow-up here.

**Comment by skynetindia** [comment]
@stpaultim Thanks for the response. We are eagerly waiting for some response. Hope someone gonna response soon

**Comment by jenlampton** [comment]
Hello @skynetindia and welcome! We will need a few things from you to join contrib.

We will need you to answer the following: 

**If you have chosen option 2 or 1 above, do you agree to the [Backdrop Contributed Project Agreement](https://github.com/backdrop-ops/contrib#backdrop-contributed-project-agreement)**


These are the requirements for joining the contrib group: 

- [ ] Include a README.md file containing license and maintainer information.
      You can use this example: https://github.com/skynettechnologies/backdrop_allinoneaccessibility/blob/main/README.md
- [x] Include a LICENSE.txt file.
      You can use this example: https://github.com/skynettechnologies/backdrop_allinoneaccessibility/blob/main/LICENSE.txt.
- [x] N/A If porting a Drupal 7 project, Maintain the Git history from Drupal.

Please see [the example README.md file provided](https://github.com/skynettechnologies/backdrop_allinoneaccessibility/blob/main/README.md) for which sections are required and which are optional. The following sections are missing:
- Issues
- Current Maintainers
- Credits
- License

### Issues
You need to include a link to the official Backdrop issue queue.

Item #7 in the [Backdrop Contributed Project Agreement states](https://github.com/backdrop-ops/contrib#backdrop-contributed-project-agreement): `Any project you create or maintain will have the GitHub issue tracker enabled for official communication.`

It may be helpful to clarify which kinds of issues you would like reported in which place. Since most our community will be using the official issue queue, you should expect to triage that list and refer people to your internal support system when a problem is beyond the scope of the code managed here. 

### Current Maintainers
Please add the Name(s) of the current maintainer(s) and link to their GitHub profile(s).  You are welcome to list `Skynet Technologies USA LLC` under the `Credits` section, but we will need to know which people we should contact in the case of a security issue. An organization is not sufficient here. 

### Credits
I suspect you can change the `About` section to `Credits` since it contains the same information. You may also want to move the links from the `Acknowledgements` section to `Documentation`, since there aren't any acknowledgements in there. :)

### License
Though you included the `License.txt` file as requested, you have not included the matching section in the readme. 

--
I'm also going to do a quick code review, but items in my next comment will be suggestions. Take or leave them as you see fit! We do this review specifically looking for things that may be helpful for those who are new to Backdrop. :)

**Comment by jenlampton** [comment]
The following is a list of suggested improvements to the module. Take or leave them as you see fit! We do this review specifically looking for things that may be helpful for those who are new to Backdrop. :)

---

## `README.md`
There are a few other issues with the README.md file that should probably also be addressed. These files are parsed and used to generate the "Project" page on backdropcms.org, so it is helpful for everyone if the markdown and content is consistent with other projects. Here are a few suggested changes:

- Remove `Backdrop CMS Module` from the name of the module.
- The heading "Description" should be removed
- Under "Prerequisites" You have indicated that people are required to be on v1.28.0. Are you sure this module depends on that core version? There are many people that are not always on the latest release. If not, you could be inadvertently preventing much of our community from using this module. (If this module does depend on the Icon API or something that was only added in 1.28.0, that Prerequisite should remain.)
- Under "Installation", remove all  the instructions on how to install a module generally, and include only what is necessary for this specific module. We recommend that Step 1 is "Install this module using the official Backdrop CMS instructions" to avoid duplicate information across projects. It looks like the only information that's not common knowledge here is what comes after Step 6. 
- Under "Configuration " It would be helpful if you could elaborate on how to "register the settings". Are there specific settings that are needed? An API key or similar? How are they "registered"? Do they need to be activated with an external service?

## `allinoneaccessibility.info`
- Remove information that was added by the Backdrop packaging script from the repository. (This information will be added automatically when you make the first release on GitHub.)
- Shorten the "tags" list. The only place these tags are used is in the "Search" box on the modules page, so jamming it with different variations of the same keywords has the same result as using each word only once. Words like web, digital, plugin, widget, site should be avoided since they are all implied. Things like "alternative" and "friendly" also add little value, and I doubt that people will be on their own website looking to enable/disable a module named "Australian DDA" or "Israeli Standard 5568". They might have a module named "userways" but if this module catches under a search for that name, they might accidentally uninstall this one, when they meant to uninstall "userways". So I would recommend avoiding that. :) Here's a more reasonable tags list:
```
tags[] = accessibility
tags[] = a11y
tags[] = WCAG
tags[] = ATAG
tags[] = ADA
tags[] = disability
tags[] = disabilities
tags[] = compliance
```

## `allinoneaccessibility.install`

This file includes an upgrade from Drupal 7. Is there a Drupal 7 version of this module? I was unable to find a release for one at https://www.drupal.org/project/all_in_one_accessibility. 
* If there is one, then this module needs to maintain the commit history of that Drupal 7 project. If it was not created from scratch for Backdrop, we need to credit the previous authors, and maintain previous change records.
* If there is not one, the function `allinoneaccessibility_update_1000()` should be removed. (If the file is empty, it can be deleted)

## `allinoneaccessibility.admin.inc`

The `<style>` tag here should be removed. How to include CSS on only the admin page is demonstrated in the [metatag module](https://github.com/backdrop-contrib/metatag/blob/1.x-1.x/metatag.admin.inc#L187). Create a new css file named `/allinoneaccessibility.admin.css`.  Locate the file using `backdrop_get_path()`. Load the file onto the admin form using `#attached` via the Form API. 

## `allinoneaccessibility.module`

The menu callback here uses the permission `administer allinoneaccessibility` but the module does not implement `hook_permission()` to define this permission. (There's a docblock for it, but the function itself is missing). Metatag also has [an example](https://github.com/backdrop-contrib/metatag/blob/1.x-1.x/metatag.module#L86) of how to implement this hook if needed.

I hope this information is helpful :)

**Comment by klonos** [comment]
@skynetindia is this the same technology as use by the https://www.drupal.org/project/all_in_one_accessibility module in Drupal, or different?
